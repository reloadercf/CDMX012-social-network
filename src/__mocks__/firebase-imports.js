/* eslint-disable no-unused-vars */
/* eslint-disable max-classes-per-file */
export const getFirestore = () => Promise.resolve({});
export const doc = (db, collection, docId) => ({ db, collection, docId });
export const setDoc = () => Promise.resolve({});
export const getDoc = jest.fn(
  () => {
    const info = {
      exists: () => false,
    };

    return Promise.resolve(info);
  },
);

const user = {
  currentUser: {
    uid: 'saraiG',
  },
};
// eslint-disable-next-line no-unused-vars
export const getAuth = (app) => user;
export const onAuthStateChanged = () => Promise.resolve({});
export const initializeApp = () => Promise.resolve({});
export const createUserWithEmailAndPassword = () => Promise.resolve({});
export const signInWithPopup = jest.fn((auth, provider) => Promise.resolve({}));
export class GoogleAuthProvider {
  constructor() {
    this.name = 'google';
  }
}
export class FacebookAuthProvider {
  constructor() {
    this.name = 'fb';
  }
}
export class GithubAuthProvider {
  constructor() {
    this.name = 'git';
  }
}
export const getAdditionalUserInfo = () => Promise.resolve({});
export const signInWithEmailAndPassword = () => Promise.resolve({});
export const deleteDoc = () => Promise.resolve({});
export const query = () => Promise.resolve({});
export const where = () => Promise.resolve({});
export const onSnapshot = () => Promise.resolve({});
export const updateDoc = () => Promise.resolve({}); // updateDoc regresa Promise<void>
export const orderBy = () => Promise.resolve({});
export const serverTimestamp = () => Promise.resolve({});
export const collection = () => Promise.resolve({});
// eslint-disable-next-line no-unused-vars
export const signOut = jest.fn((auth) => Promise.resolve());
