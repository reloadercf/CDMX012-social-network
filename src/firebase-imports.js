/* eslint-disable object-curly-newline */
/* eslint-disable import/no-unresolved */
// eslint-disable-next-line import/no-unresolved
export { getFirestore, doc, setDoc, getDoc, collection, addDoc, query, where, onSnapshot, deleteDoc, updateDoc, orderBy, serverTimestamp } from 'https://www.gstatic.com/firebasejs/9.6.6/firebase-firestore.js';
export { getAuth, onAuthStateChanged, signOut } from 'https://www.gstatic.com/firebasejs/9.6.6/firebase-auth.js';
export { createUserWithEmailAndPassword, signInWithPopup, GoogleAuthProvider, FacebookAuthProvider, GithubAuthProvider, getAdditionalUserInfo, signInWithEmailAndPassword } from 'https://www.gstatic.com/firebasejs/9.6.6/firebase-auth.js';
export { initializeApp } from 'https://www.gstatic.com/firebasejs/9.6.6/firebase-app.js';
