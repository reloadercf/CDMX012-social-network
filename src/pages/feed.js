/* eslint-disable import/no-cycle */
import { saveNewPostData, getPosts } from '../lib/posts.js';
import { slideshow } from '../components/slideshow.js';
import { showAndHideItems, goToTop } from '../components/ui.js';
import { menu } from '../components/menu.js';

export const feed = () => {
  const readingPage = document.createElement('div');
  readingPage.setAttribute('class', 'readingPage');

  // Header section
  const header = document.createElement('header');
  header.setAttribute('class', 'feed-header');

  const logoTitle = document.createElement('div');
  logoTitle.setAttribute('class', 'logo-title');

  const logo = document.createElement('img');
  logo.setAttribute('class', 'logo-book-feed');
  logo.setAttribute('src', './assets/logo sin fondo 1.png');
  logo.setAttribute('alt', 'logo book reads');

  const bookreads = document.createElement('img');
  bookreads.setAttribute('class', 'titleBookReads-feed');
  bookreads.setAttribute('src', './assets/bookreads-white-logo.png');
  bookreads.setAttribute('alt', 'titleBookReads');

  logoTitle.append(logo, bookreads);

  const configMenu = document.createElement('img');
  configMenu.setAttribute('class', 'config-menu');
  configMenu.setAttribute('src', './assets/white-config-icon.png');
  configMenu.setAttribute('alt', 'configuration menu');

  configMenu.addEventListener('click', () => {
    const body = document.body;
    body.style.overflow = 'hidden';
    readingPage.append(menu());
  });

  header.append(logoTitle, configMenu);

  // feed content section
  const feedContent = document.createElement('div');
  feedContent.setAttribute('class', 'feed-content');

  // book suggestions h2
  const suggestions = document.createElement('div');
  const bookSuggest = document.createElement('h2');
  bookSuggest.setAttribute('class', 'bookSuggest');
  bookSuggest.innerHTML = 'Book suggestions:';
  suggestions.append(bookSuggest);

  // Create a new post section
  const makeNewPost = document.createElement('div');
  makeNewPost.setAttribute('class', 'make-new-post');
  makeNewPost.setAttribute('id', 'makeNewPost');
  makeNewPost.innerHTML = "What's on your mind?";

  const readingForm = document.createElement('form');
  readingForm.setAttribute('class', 'reading-form');
  readingForm.setAttribute('id', 'readingForm');

  const insertReadingTitle = document.createElement('div');
  insertReadingTitle.setAttribute('class', 'reading-area');

  const readingTitle = document.createElement('div');
  readingTitle.innerHTML = '<label for="reading" class="reading">Reading:</label>';

  const readingBook = document.createElement('input');
  readingBook.setAttribute('class', 'book-title');
  readingBook.setAttribute('id', 'bookTitle');
  readingBook.setAttribute('type', 'text');
  readingBook.setAttribute('placeholder', 'What are you reading?');
  readingBook.setAttribute('name', 'bookTitle');

  insertReadingTitle.append(readingTitle, readingBook);

  const readingDescription = document.createElement('textarea');
  readingDescription.setAttribute('class', 'post-content');
  readingDescription.setAttribute('name', 'postContent');
  readingDescription.setAttribute('placeholder', "What's on your mind?");

  const newPostBtn = document.createElement('input');
  newPostBtn.setAttribute('type', 'submit');
  newPostBtn.setAttribute('value', 'Share');
  newPostBtn.setAttribute('class', 'new-post-button');
  newPostBtn.setAttribute('id', 'newPostButton');

  readingForm.append(insertReadingTitle, readingDescription, newPostBtn);

  // Posts section
  const postsArea = document.createElement('div');
  postsArea.setAttribute('class', 'posts');
  postsArea.setAttribute('id', 'postsArea');

  const slideshowElement = slideshow();
  feedContent.append(suggestions, slideshowElement, makeNewPost, readingForm, postsArea);
  readingPage.append(header, feedContent);

  document.addEventListener('DOMContentLoaded', getPosts()); // Averiguar cómo ordenar los posts, más reciente primero

  logoTitle.addEventListener('click', () => {
    goToTop();
  });

  makeNewPost.addEventListener('click', () => {
    showAndHideItems(readingForm, makeNewPost);
  });

  newPostBtn.addEventListener('click', (e) => {
    e.preventDefault();
    saveNewPostData(readingForm);
    showAndHideItems(makeNewPost, readingForm);
    console.error('una vez');
  });

  return readingPage;
};
