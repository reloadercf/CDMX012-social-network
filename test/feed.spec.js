/**
 * @jest-environment jsdom
 */
import { feed } from '../src/pages/feed.js';
import { getDoc } from '../src/firebase-imports.js';

jest.mock('../src/firebase-imports.js');

describe('Tests of feed', () => {
  test('render feed', () => {
    const renderFeed = feed();
    document.body.innerHTML = '<div id="root"></div>';
    const rootDiv = document.getElementById('root');
    rootDiv.appendChild(renderFeed);
    expect(rootDiv.innerHTML).toMatchSnapshot();
  });
});

describe('Hamburger menu', () => {
  const feedPage = feed();

  it('Hamburger menu should not exist if hamburger menu is not clicked', () => {
    const hasChild = feedPage.querySelector('#general-container') == null;
    expect(hasChild).toEqual(true);
  });

  it('When clicking hamburger menu it should open the menu', () => {
    getDoc.mockResolvedValue({
      exists: () => true,
      data: () => ({
        name: 'Name1',
        username: 'username1',
        bio: 'texto',
      }),
    });
    const menu = feedPage.querySelector('.config-menu');
    menu.dispatchEvent(new Event('click'));
    const hasChild = feedPage.querySelector('#general-container') != null;
    expect(hasChild).toEqual(true);
  });
});
