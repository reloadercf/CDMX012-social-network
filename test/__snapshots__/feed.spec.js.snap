// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of feed render feed 1`] = `"<div class=\\"readingPage\\"><header class=\\"feed-header\\"><div class=\\"logo-title\\"><img class=\\"logo-book-feed\\" src=\\"./assets/logo sin fondo 1.png\\" alt=\\"logo book reads\\"><img class=\\"titleBookReads-feed\\" src=\\"./assets/bookreads-white-logo.png\\" alt=\\"titleBookReads\\"></div><img class=\\"config-menu\\" src=\\"./assets/white-config-icon.png\\" alt=\\"configuration menu\\"></header><div class=\\"feed-content\\"><div><h2 class=\\"bookSuggest\\">Book suggestions:</h2></div><div class=\\"slideshow-container\\"><div class=\\"slide fade\\" style=\\"display: block;\\"><img src=\\"./assets/book1.png\\" alt=\\"BookOne\\"></div><div class=\\"slide fade\\" style=\\"display: none;\\"><img src=\\"./assets/book2.png\\" alt=\\"BookTwo\\"></div><div class=\\"slide fade\\" style=\\"display: none;\\"><img src=\\"./assets/book3.png\\" alt=\\"BookThree\\"></div><div class=\\"slide fade\\" style=\\"display: none;\\"><img src=\\"./assets/book4.png\\" alt=\\"BookFour\\"></div><a class=\\"btnPrev\\" id=\\"btnPrev\\">❮</a><a class=\\"btnNext\\" id=\\"btnNext\\">❯</a></div><div class=\\"make-new-post\\" id=\\"makeNewPost\\">What's on your mind?</div><form class=\\"reading-form\\" id=\\"readingForm\\"><div class=\\"reading-area\\"><div><label for=\\"reading\\" class=\\"reading\\">Reading:</label></div><input class=\\"book-title\\" id=\\"bookTitle\\" type=\\"text\\" placeholder=\\"What are you reading?\\" name=\\"bookTitle\\"></div><textarea class=\\"post-content\\" name=\\"postContent\\" placeholder=\\"What's on your mind?\\"></textarea><input type=\\"button\\" value=\\"Share\\" class=\\"new-post-button\\" id=\\"newPostButton\\"></form><div class=\\"posts\\" id=\\"postsArea\\"></div></div></div>"`;
